File: app/src/main/java/com/example/ui/screens/ProfileScreen.kt
package com.example.ui.screens

import androidx.compose.foundation.background
import androidx.compose.foundation.layout.*
import androidx.compose.foundation.rememberScrollState
import androidx.compose.foundation.shape.RoundedCornerShape
import androidx.compose.foundation.verticalScroll
import androidx.compose.material.icons.Icons
import androidx.compose.material.icons.filled.Save
import androidx.compose.material.icons.outlined.Badge
import androidx.compose.material.icons.outlined.ContactMail
import androidx.compose.material.icons.outlined.HistoryEdu
import androidx.compose.material.icons.outlined.Person
import androidx.compose.material3.*
import androidx.compose.runtime.*
import androidx.compose.ui.Alignment
import androidx.compose.ui.Modifier
import androidx.compose.ui.platform.testTag
import androidx.compose.ui.text.font.FontWeight
import androidx.compose.ui.unit.dp
import com.example.data.UserProfileEntity
import com.example.viewmodel.MainViewModel
import kotlinx.coroutines.launch

@OptIn(ExperimentalMaterial3Api::class)
@Composable
fun ProfileScreen(
    viewModel: MainViewModel,
    modifier: Modifier = Modifier
) {
    val savedProfile by viewModel.userProfile.collectAsState()

    var name by remember { mutableStateOf("") }
    var email by remember { mutableStateOf("") }
    var phone by remember { mutableStateOf("") }
    var title by remember { mutableStateOf("") }
    var skills by remember { mutableStateOf("") }
    var experience by remember { mutableStateOf("") }
    var education by remember { mutableStateOf("") }
    var bio by remember { mutableStateOf("") }

    val snackbarHostState = remember { SnackbarHostState() }
    val scope = rememberCoroutineScope()

    // Sync state with db loaded value
    LaunchedEffect(savedProfile) {
        name = savedProfile.fullName
        email = savedProfile.email
        phone = savedProfile.phone
        title = savedProfile.targetTitle
        skills = savedProfile.skills
        experience = savedProfile.experience
        education = savedProfile.education
        bio = savedProfile.bio
    }

    Scaffold(
        snackbarHost = { SnackbarHost(snackbarHostState) },
        floatingActionButton = {
            ExtendedFloatingActionButton(
                onClick = {
                    val updatedProfile = UserProfileEntity(
                        id = 1,
                        fullName = name,
                        email = email,
                        phone = phone,
                        targetTitle = title,
                        skills = skills,
                        experience = experience,
                        education = education,
                        bio = bio
                    )
                    viewModel.saveProfile(updatedProfile)
                    scope.launch {
                        snackbarHostState.showSnackbar(
                            message = "Profile Synced & Ready for AI Tools! ✅",
                            duration = SnackbarDuration.Short
                        )
                    }
                },
                modifier = Modifier.testTag("save_profile_fab"),
                icon = { Icon(Icons.Default.Save, "Save") },
                text = { Text("Sync Profile") },
                containerColor = MaterialTheme.colorScheme.primary,
                contentColor = MaterialTheme.colorScheme.onPrimary
            )
        },
        modifier = modifier.fillMaxSize()
    ) { innerPadding ->
        Column(
            modifier = Modifier
                .padding(innerPadding)
                .fillMaxSize()
                .background(MaterialTheme.colorScheme.background)
                .verticalScroll(rememberScrollState())
                .padding(16.dp),
            verticalArrangement = Arrangement.spacedBy(16.dp)
        ) {
            // Header card
            Card(
                modifier = Modifier.fillMaxWidth(),
                colors = CardDefaults.cardColors(containerColor = MaterialTheme.colorScheme.surfaceVariant.copy(alpha = 0.5f))
            ) {
                Column(modifier = Modifier.padding(16.dp)) {
                    Text(
                        text = "Your Offline Resume Vault",
                        style = MaterialTheme.typography.titleMedium,
                        fontWeight = FontWeight.Bold,
                        color = MaterialTheme.colorScheme.primary
                    )
                    Spacer(modifier = Modifier.height(4.dp))
                    Text(
                        text = "Fill in your background to enable instant AI cover letter composing, resume critiquing, and personalized interview practice templates.",
                        style = MaterialTheme.typography.bodyMedium,
                        color = MaterialTheme.colorScheme.onSurfaceVariant
                    )
                }
            }

            // Section: Contact details
            ProfileSectionHeader(icon = Icons.Outlined.Person, title = "Personal Details")

            OutlinedTextField(
                value = name,
                onValueChange = { name = it },
                label = { Text("Full Name") },
                placeholder = { Text("e.g. Alex Mercer") },
                modifier = Modifier
                    .fillMaxWidth()
                    .testTag("profile_name_input"),
                singleLine = true
            )

            OutlinedTextField(
                value = title,
                onValueChange = { title = it },
                label = { Text("Target Professional Title") },
                placeholder = { Text("e.g. Senior Software Engineer") },
                modifier = Modifier
                    .fillMaxWidth()
                    .testTag("profile_title_input"),
                singleLine = true
            )

            Row(
                modifier = Modifier.fillMaxWidth(),
                horizontalArrangement = Arrangement.spacedBy(12.dp)
            ) {
                OutlinedTextField(
                    value = email,
                    onValueChange = { email = it },
                    label = { Text("Email Address") },
                    placeholder = { Text("alex@mercer.io") },
                    modifier = Modifier
                        .weight(1f)
                        .testTag("profile_email_input"),
                    singleLine = true
                )

                OutlinedTextField(
                    value = phone,
                    onValueChange = { phone = it },
                    label = { Text("Phone") },
                    placeholder = { Text("+1 (555) 0199") },
                    modifier = Modifier
                        .weight(1f)
                        .testTag("profile_phone_input"),
                    singleLine = true
                )
            }

            // Section: Resume Core Details
            ProfileSectionHeader(icon = Icons.Outlined.ContactMail, title = "Skills & Profile")

            OutlinedTextField(
                value = skills,
                onValueChange = { skills = it },
                label = { Text("Core Skills (Comma-separated)") },
                placeholder = { Text("Kotlin, Jetpack Compose, REST APIs, Room DB, Git, Python, SQL") },
                modifier = Modifier
                    .fillMaxWidth()
                    .testTag("profile_skills_input")
            )

            OutlinedTextField(
                value = bio,
                onValueChange = { bio = it },
                label = { Text("Professional Summary / Bio") },
                placeholder = { Text("Enthusiastic and results-driven software professional with 3+ years experience designing fluid customer interfaces...") },
                modifier = Modifier
                    .fillMaxWidth()
                    .height(100.dp)
                    .testTag("profile_bio_input"),
                maxLines = 4
            )

            // Section: Work Experience and Education
            ProfileSectionHeader(icon = Icons.Outlined.HistoryEdu, title = "Experience & Education")

            OutlinedTextField(
                value = experience,
                onValueChange = { experience = it },
                label = { Text("Work Experience") },
                placeholder = { Text("• Senior Mobile Dev at NovaTech (2024-Present): Lead Android compostable UI system overhaul.\n• Software Eng at ByteSized Corp (2022-2024): Built API integrations.") },
                modifier = Modifier
                    .fillMaxWidth()
                    .height(120.dp)
                    .testTag("profile_experience_input"),
                maxLines = 6
            )

            OutlinedTextField(
                value = education,
                onValueChange = { education = it },
                label = { Text("Education Summary") },
                placeholder = { Text("B.S. in Computer Science - University of Texas at Austin (2021)") },
                modifier = Modifier
                    .fillMaxWidth()
                    .testTag("profile_education_input"),
                singleLine = true
            )

            Spacer(modifier = Modifier.height(64.dp)) // Padding to stay clear of the FAB
        }
    }
}

@Composable
fun ProfileSectionHeader(
    icon: androidx.compose.ui.graphics.vector.ImageVector,
    title: String
) {
    Row(
        verticalAlignment = Alignment.CenterVertically,
        modifier = Modifier.fillMaxWidth()
    ) {
        Icon(
            imageVector = icon,
            contentDescription = null,
            tint = MaterialTheme.colorScheme.secondary,
            modifier = Modifier.size(22.dp)
        )
        Spacer(modifier = Modifier.width(8.dp))
        Text(
            text = title,
            style = MaterialTheme.typography.titleMedium,
            fontWeight = FontWeight.Bold,
            color = MaterialTheme.colorScheme.onBackground
        )
    }
}

File: lib/screens/profile_screen.dart
import 'package:flutter/material.dart';
import 'package:provider/provider.dart';
import '../providers/jobs_provider.dart';
import '../models/models.dart';

class ProfileScreen extends StatefulWidget {
  const ProfileScreen({super.key});

  @override
  State<ProfileScreen> createState() => _ProfileScreenState();
}

class _ProfileScreenState extends State<ProfileScreen> {
  final _formKey = GlobalKey<FormState>();

  late TextEditingController _nameController;
  late TextEditingController _titleController;
  late TextEditingController _emailController;
  late TextEditingController _phoneController;
  late TextEditingController _skillsController;
  late TextEditingController _bioController;
  late TextEditingController _experienceController;
  late TextEditingController _educationController;

  @override
  void initState() {
    super.initState();
    final profile = Provider.of<JobsProvider>(context, listen: false).userProfile;

    _nameController = TextEditingController(text: profile.fullName);
    _titleController = TextEditingController(text: profile.targetTitle);
    _emailController = TextEditingController(text: profile.email);
    _phoneController = TextEditingController(text: profile.phone);
    _skillsController = TextEditingController(text: profile.skills);
    _bioController = TextEditingController(text: profile.bio);
    _experienceController = TextEditingController(text: profile.experience);
    _educationController = TextEditingController(text: profile.education);
  }

  @override
  void dispose() {
    _nameController.dispose();
    _titleController.dispose();
    _emailController.dispose();
    _phoneController.dispose();
    _skillsController.dispose();
    _bioController.dispose();
    _experienceController.dispose();
    _educationController.dispose();
    super.dispose();
  }

  void _saveProfile() {
    final provider = Provider.of<JobsProvider>(context, listen: false);
    final updated = UserProfileEntity(
      id: 1,
      fullName: _nameController.text,
      email: _emailController.text,
      phone: _phoneController.text,
      targetTitle: _titleController.text,
      skills: _skillsController.text,
      experience: _experienceController.text,
      education: _educationController.text,
      bio: _bioController.text,
    );
    provider.saveProfile(updated);

    ScaffoldMessenger.of(context).showSnackBar(
      const SnackBar(
        content: Text('Profile Synced & Ready for AI Tools! ✅'),
        backgroundColor: Color(0xFF10B981),
        duration: Duration(seconds: 2),
      ),
    );
  }

  @override
  Widget build(BuildContext context) {
    return Scaffold(
      backgroundColor: const Color(0xFF0F172A),
      appBar: AppBar(
        title: const Text('My Resume Vault', style: TextStyle(fontWeight: FontWeight.bold)),
      ),
      floatingActionButton: FloatingActionButton.extended(
        onPressed: _saveProfile,
        backgroundColor: const Color(0xFF3B82F6),
        foregroundColor: Colors.white,
        icon: const Icon(Icons.sync),
        label: const Text('Sync Profile', style: TextStyle(fontWeight: FontWeight.bold)),
      ),
      body: SingleChildScrollView(
        padding: const EdgeInsets.only(left: 16, right: 16, top: 16, bottom: 80),
        child: Form(
          key: _formKey,
          child: Column(
            crossAxisAlignment: CrossAxisAlignment.start,
            children: [
              // Info Card
              Card(
                color: const Color(0xFF1E293B).withOpacity(0.5),
                shape: RoundedRectangleBorder(
                  borderRadius: BorderRadius.circular(16),
                  side: BorderSide(color: const Color(0xFF334155).withOpacity(0.5)),
                ),
                child: Padding(
                  padding: const EdgeInsets.all(16.0),
                  child: Column(
                    crossAxisAlignment: CrossAxisAlignment.start,
                    children: [
                      const Text(
                        'Your Offline Resume Vault',
                        style: TextStyle(color: Color(0xFF3B82F6), fontWeight: FontWeight.bold, fontSize: 16),
                      ),
                      const SizedBox(height: 6),
                      const Text(
                        'Fill in your background to enable instant AI cover letter composing, resume critiquing, and personalized interview practice templates.',
                        style: TextStyle(color: Color(0xFF94A3B8), fontSize: 13, height: 1.5),
                      ),
                    ],
                  ),
                ),
              ),
              const SizedBox(height: 20),

              // Section 1: Personal Details
              _buildSectionHeader(Icons.person_outline, 'Personal Details'),
              const SizedBox(height: 12),

              _buildTextField(
                controller: _nameController,
                label: 'Full Name',
                hint: 'e.g. Alex Mercer',
              ),
              const SizedBox(height: 12),

              _buildTextField(
                controller: _titleController,
                label: 'Target Professional Title',
                hint: 'e.g. Senior Software Engineer',
              ),
              const SizedBox(height: 12),

              Row(
                children: [
                  Expanded(
                    child: _buildTextField(
                      controller: _emailController,
                      label: 'Email Address',
                      hint: 'alex@mercer.io',
                      keyboardType: TextInputType.emailAddress,
                    ),
                  ),
                  const SizedBox(width: 12),
                  Expanded(
                    child: _buildTextField(
                      controller: _phoneController,
                      label: 'Phone Number',
                      hint: '+1 (555) 0199',
                      keyboardType: TextInputType.phone,
                    ),
                  ),
                ],
              ),
              const SizedBox(height: 24),

              // Section 2: Skills & Bio
              _buildSectionHeader(Icons.business_center_outlined, 'Skills & Profile Summary'),
              const SizedBox(height: 12),

              _buildTextField(
                controller: _skillsController,
                label: 'Core Skills (Comma-separated)',
                hint: 'Kotlin, Jetpack Compose, REST APIs, Room DB, Git, Python, SQL',
              ),
              const SizedBox(height: 12),

              _buildTextField(
                controller: _bioController,
                label: 'Professional Summary / Bio',
                hint: 'Enthusiastic and results-driven software professional with 3+ years experience...',
                maxLines: 4,
              ),
              const SizedBox(height: 24),

              // Section 3: Experience & Education
              _buildSectionHeader(Icons.school_outlined, 'Experience & Education'),
              const SizedBox(height: 12),

              _buildTextField(
                controller: _experienceController,
                label: 'Work Experience',
                hint: '• Senior Mobile Dev at NovaTech (2024-Present)\n• Software Eng at ByteSized Corp (2022-2024)',
                maxLines: 5,
              ),
              const SizedBox(height: 12),

              _buildTextField(
                controller: _educationController,
                label: 'Education Summary',
                hint: 'B.S. in Computer Science - University of Texas at Austin (2021)',
              ),
            ],
          ),
        ),
      ),
    );
  }

  Widget _buildSectionHeader(IconData icon, String title) {
    return Row(
      children: [
        Icon(icon, color: const Color(0xFFA78BFA), size: 20),
        const SizedBox(width: 8),
        Text(
          title,
          style: const TextStyle(color: Colors.white, fontWeight: FontWeight.bold, fontSize: 16),
        ),
      ],
    );
  }

  Widget _buildTextField({
    required TextEditingController controller,
    required String label,
    required String hint,
    int maxLines = 1,
    TextInputType keyboardType = TextInputType.text,
  }) {
    return TextField(
      controller: controller,
      maxLines: maxLines,
      keyboardType: keyboardType,
      style: const TextStyle(color: Colors.white, fontSize: 14),
      decoration: InputDecoration(
        labelText: label,
        labelStyle: const TextStyle(color: Color(0xFF64748B), fontSize: 13),
        hintText: hint,
        hintStyle: const TextStyle(color: Color(0xFF334155), fontSize: 13),
        filled: true,
        fillColor: const Color(0xFF1E293B).withOpacity(0.3),
        focusedBorder: OutlineInputBorder(
          borderRadius: BorderRadius.circular(12),
          borderSide: const BorderSide(color: Color(0xFF3B82F6), width: 1.5),
        ),
        enabledBorder: OutlineInputBorder(
          borderRadius: BorderRadius.circular(12),
          borderSide: const BorderSide(color: Color(0xFF334155)),
        ),
        alignLabelWithHint: true,
      ),
    );
  }
}
